### PR TITLE
Document macros defined by cmakedefine

### PR DESCRIPTION
**Pull request by masterleinad**

### CODE REVIEW

File: API-Reference/Macros.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+| `KOKKOS_ARCH_AMPERE80` | Optimize for NVIDIA Ampere generation CC 8.0 (GPU) |
+| `KOKKOS_ARCH_AMPERE86 ` | Optimize for NVIDIA Ampere generation CC 8.6 (GPU) |
+| `KOKKOS_ARCH_AMD_ZEN` | Optimize for AMD Zen architecture (HOST) |
+| `KOKKOS_ARCH_AMD_ZEN2` | Optimize for AMD Zen2 architecture (HOST) |
+| `KOKKOS_ARCH_AMD_ZEN3` | Optimize for AMD Zen3 architecture (HOST) |
+| `KOKKOS_ARCH_VEGA` | Set if any AMD Vega GPU architecture as been enabled |
+| `KOKKOS_ARCH_VEGA900` | Optimize for AMD GPU MI25 GFX900 (GPU) |
+| `KOKKOS_ARCH_VEGA906` | Optimize for AMD GPU MI50/MI60 GFX906 (GPU) |
+| `KOKKOS_ARCH_VEGA908` | Optimize for AMD GPU MI100 GFX908 (GPU) |
+| `KOKKOS_ARCH_VEGA90A` | Optimize for AMD GPU MI200 GFX90A (GPU) |
```
Comment (dalg24) [inline API-Reference/Macros.md:132]: @Rombur should we say "MI200 series" to encompass MI210, MI250, and MI250X

File: API-Reference/Macros.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+| Macro | Description |
+| ----- | ----------- |
+| `KOKKOS_VERSION` | The Kokkos version; `KOKKOS_VERSION % 100` is the patch level, `KOKKOS_VERSION / 100 % 100` is the minor version, and `KOKKOS_VERSION / 10000` is the major version. |
+
+## General Settings
+
+| Macro | Description |
+| ------| ---------------|
+| `KOKKOS_ENABLE_DEBUG` | Defined if extra debug features are activated. |
+| `KOKKOS_ENABLE_DEBUG_BOUNDS_CHECK` | Defined if extra bounds checks are enabled/|
```
Comment (dalg24) [inline API-Reference/Macros.md:14]:
```suggestion
| `KOKKOS_ENABLE_DEBUG_BOUNDS_CHECK` | Defined if extra bounds checks are enabled. |
```

File: API-Reference/Macros.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+| `KOKKOS_ENABLE_DEBUG_DUALVIEW_MODIFY_CHECK` | Defined if debug checks for Kokkos::DualView objects are enabled. |
```
Comment (dalg24) [inline API-Reference/Macros.md:15]:
```suggestion
| `KOKKOS_ENABLE_DEBUG_DUALVIEW_MODIFY_CHECK` | Defined if debug checks for `Kokkos::DualView` objects are enabled. |
```

File: API-Reference/Macros.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+| `KOKKOS_ENABLE_DEPRECATED_CODE_3` | Defined if features deprecated in major release 3 are still available. |
+| `KOKKOS_ENABLE_DEPRECATION_WARNING` | Defined if deprecated features generate deprecation warnings |
+| `KOKKOS_ENABLE_COMPILER_WARNINGS` | A subset of compiler warnings are enabled for Kokkos. |
```
Comment (dalg24) [inline API-Reference/Macros.md:18]: I did not realize we were defining that.  It feels like an oversight.  Can anyone think of a good reason to have it?

File: API-Reference/Macros.md [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+| `KOKKOS_ENABLE_PROFILING_LOAD_PRINT` | Kokkos will output a message when the profiling library is loaded. |
+| `KOKKOS_ENABLE_TUNING` | Whether bindings for tunings are available (see [#2422](https://github.com/kokkos/kokkos/pull/2422)). |
+| `KOKKOS_ENABLE_LARGE_MEM_TESTS` | Whether large memory tests are enabled (only applies to some unit tests). |
```
Comment (dalg24) [inline API-Reference/Macros.md:21]: I don't think this one would be advertised.

File: API-Reference/Macros.md (hunk 1 above)
Comment (Rombur) [reply API-Reference/Macros.md:132]: yes, we can change it to series

File: API-Reference/Macros.md [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+| Macro | Description |
+| ----- | ----------- |
+| `KOKKOS_VERSION` | The Kokkos version; `KOKKOS_VERSION % 100` is the patch level, `KOKKOS_VERSION / 100 % 100` is the minor version, and `KOKKOS_VERSION / 10000` is the major version. |
+
+## General Settings
+
+| Macro | Description |
+| ------| ---------------|
+| `KOKKOS_ENABLE_DEBUG` | Defined if extra debug features are activated. |
+| `KOKKOS_ENABLE_DEBUG_BOUNDS_CHECK` | Defined if extra bounds checks are enabled |
```
Comment (dalg24) [inline API-Reference/Macros.md:14]: Every single other entry in the table is terminated with a "."

Comment (masterleinad) [inline API-Reference/Macros.md:14]:
```suggestion
| `KOKKOS_ENABLE_DEBUG_BOUNDS_CHECK` | Defined if extra bounds checks are enabled. |
```

File: API-Reference/Macros.md [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+| `KOKKOS_ENABLE_DEBUG_DUALVIEW_MODIFY_CHECK` | Defined if debug checks for `Kokkos::DualView` objects are enabled. |
+| `KOKKOS_ENABLE_DEPRECATED_CODE_3` | Defined if features deprecated in major release 3 are still available. |
+| `KOKKOS_ENABLE_DEPRECATION_WARNING` | Defined if deprecated features generate deprecation warnings |
```
Comment (masterleinad) [inline API-Reference/Macros.md:17]:
```suggestion
| `KOKKOS_ENABLE_DEPRECATION_WARNING` | Defined if deprecated features generate deprecation warnings. |
```

File: API-Reference/Macros.md [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+| `KOKKOS_ENABLE_HPX`          | Defined if the [`HPX`](Kokkos%3A%3AHPX) execution space is enabled.         |
+| `KOKKOS_ENABLE_MEMKIND`      | Defined if the experimental `HBWSpace` execution space is enabled.          |
+| `KOKKOS_ENABLE_SYCL`         | Defined if the experimental `SYCL` execution space is enabled.              |
+
+## Backend options
+
+| Macro                        | Description                                                                 |
+| ---------------------------- | --------------------------------------------------------------------------- |
+| `KOKKOS_ENABLE_CUDA_CONSTEXPR` | Defined if the CUDA backend supports constexpr functions. |
+| `KOKKOS_ENABLE_CUDA_LAMBDA`                  | Defined if the CUDA backend supports lambdas.
```
Comment (masterleinad) [inline API-Reference/Macros.md:45]:
```suggestion
| `KOKKOS_ENABLE_CUDA_LAMBDA`                  | Defined if the CUDA backend supports lambdas. |
```

File: API-Reference/Macros.md [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+| `KOKKOS_ENABLE_CUDA_LDG_INTRINSINCS`         | Defined if the CUDA backend supports LDG intrinsics.
```
Comment (masterleinad) [inline API-Reference/Macros.md:46]:
```suggestion
| `KOKKOS_ENABLE_CUDA_LDG_INTRINSINCS`         | Defined if the CUDA backend supports LDG intrinsic. |
```

File: API-Reference/Macros.md [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+| Macro                        | Description                                                                 |
+| ---------------------------- | --------------------------------------------------------------------------- |
+| `KOKKOS_ENABLE_SERIAL`       | Defined if the [`Serial`](Kokkos%3A%3ASerial) execution space is enabled.   |
+| `KOKKOS_ENABLE_OPENMP`       | Defined if the [`OpenMP`](Kokkos%3A%3AOpenMP) execution space is enabled.   |
+| `KOKKOS_ENABLE_OPENMPTARGET` | Defined if the experimental `OpenMPTarget` execution space is enabled.      |
+| `KOKKOS_ENABLE_THREADS`      | Defined if the [`Threads`](Kokkos%3A%3AThreads) execution space is enabled. |
+| `KOKKOS_ENABLE_CUDA`         | Defined if the [`Cuda`](Kokkos%3A%3ACuda) execution space is enabled.       |
+| `KOKKOS_ENABLE_HIP`          | Defined if the experimental `HIP` execution space is enabled.               |
+| `KOKKOS_ENABLE_HPX`          | Defined if the [`HPX`](Kokkos%3A%3AHPX) execution space is enabled.         |
+| `KOKKOS_ENABLE_MEMKIND`      | Defined if the experimental `HBWSpace` execution space is enabled.          |
```
Comment (dalg24) [inline API-Reference/Macros.md:37]: This is not an execution space is it?

Comment (masterleinad) [reply API-Reference/Macros.md:37]: No, it's a memory space.

Comment (dalg24) [reply API-Reference/Macros.md:37]: So plz fix the comment.

Comment (masterleinad) [reply API-Reference/Macros.md]: 8c0c54492b73bb11fed8f4eaf9fdd74b87960469

Comment (nmm0) [reply API-Reference/Macros.md]: Thanks for fixing, wasn't 100% sure how it was supposed to be documented.

**Review by dalg24 — APPROVED** [review]

**Review by crtrott — APPROVED** [review]

**Pull request merged**